include/swift-c/DependencyScan/DependencyScan.h

```diff
@@ -0,0 +1,228 @@
+//===--- DependencyScan.h - C API for Swift Dependency Scanning ---*- C -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This C API is primarily intended to serve as the Swift Driver's
+// dependency scanning facility (https://github.com/apple/swift-driver).
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_C_DEPENDENCY_SCAN_H
+#define SWIFT_C_DEPENDENCY_SCAN_H
+
+#include "DSString.h"
+#include "DependencyScanMacros.h"
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/// The version constants for the SwiftDependencyScan C API.
+/// DEPSCAN_VERSION_MINOR should increase when there are API additions.
+/// DEPSCAN_VERSION_MAJOR is intended for "major" source/ABI breaking changes.
+#define DEPSCAN_VERSION_MAJOR 0
+#define DEPSCAN_VERSION_MINOR 1
+
+DEPSCAN_BEGIN_DECLS
+
+//=== Public Scanner Data Types -------------------------------------------===//
+
+typedef enum {
+  DEPSCAN_DEPENDENCY_INFO_SWIFT_TEXTUAL = 0,
+  DEPSCAN_DEPENDENCY_INFO_SWIFT_BINARY = 1,
+  DEPSCAN_DEPENDENCY_INFO_SWIFT_PLACEHOLDER = 2,
+  DEPSCAN_DEPENDENCY_INFO_CLANG = 3
+} ds_dependency_info_kind_t;
+
+/// Opaque container of the details specific to a given module dependency.
+typedef void *ds_module_details_t;
+
+typedef struct {
+  /// The module's name
+  /// The format is:
+  /// `<module-kind>:<module-name>`
+  /// where `module-kind` is one of:
+  /// "swiftTextual"
+  /// "swiftBinary"
+  /// "swiftPlaceholder"
+  /// "clang""
+  ds_string_t module_name;
+
+  /// The path for the module.
+  ds_string_t module_path;
+
+  /// The source files used to build this module.
+  ds_string_set_t *source_files;
+
+  /**
+   * The list of modules which this module direct depends on.
+   * The format is:
+   * `<module-kind>:<module-name>`
+   */
+  ds_string_set_t *direct_dependencies;
+
+  /// Specific details of a particular kind of module.
+  ds_module_details_t details;
+} ds_dependency_info_t;
```

Comment:
I would recommend making `ds_dependency_info_t` opaque with accessors for all the fields you want to expose.  It seems likely that you might some day want to add more data here.

Reply:
Turned into an opaque type, thank you.